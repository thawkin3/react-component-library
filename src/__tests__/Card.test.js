import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Card from '../Card';

let props;
let wrapper;

describe('<Card /> component', () => {
    beforeEach(() => {
        props = {
            children: <div><p>Child paragraph 1</p><p>Child paragraph 2</p></div>,
            childrenClasses: undefined,
            childrenStyles: undefined,
            headerClasses: undefined,
            headerStyles: undefined,
            headerText: 'This is the header',
        };
        wrapper = shallow(<Card {...props} />);
    });

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card {...props} />, div);
    });

    it('matches the snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders an h1 tag', () => {
        expect(wrapper.find('h1').length).toEqual(1);
    });

    it('renders a div with the class "react-component-library__card"', () => {
        expect(wrapper.find('div.react-component-library__card').length).toEqual(1);
    });
});
