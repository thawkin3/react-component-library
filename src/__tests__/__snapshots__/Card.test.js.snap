// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> component matches the snapshot 1`] = `
<div
  className="react-component-library__card "
  style={Object {}}
>
  <h1
    className="react-component-library__card__header "
    style={Object {}}
  >
    This is the header
  </h1>
  <div
    className="react-component-library__card__children "
    style={Object {}}
  >
    <div>
      <p>
        Child paragraph 1
      </p>
      <p>
        Child paragraph 2
      </p>
    </div>
  </div>
</div>
`;
